server/setup_flow.go

```diff
@@ -554,15 +584,51 @@ func (p *Plugin) submitCreateCloudInstance(f *flow.Flow, submission map[string]i
 		jiraURL = fmt.Sprintf("https://%s.atlassian.net", jiraURL)
 	}
 
-	jiraURL, err := p.installInactiveCloudInstance(jiraURL, f.UserID)
+	jiraURL, instance, err := p.installCloudOAuthInstance(jiraURL, "", "")
+	if err != nil {
+		return "", nil, nil, err
+	}
+
+	return stepCloudOAuthConfigure, flow.State{
+		keyEdition:          string(CloudOAuthInstanceType),
+		keyJiraURL:          jiraURL,
+		keyInstance:         instance,
+		keyOAuthCompleteURL: p.GetPluginURL() + instancePath(routeOAuth2Complete, types.ID(jiraURL)),
+		keyConnectURL:       p.GetPluginURL() + instancePath(routeUserConnect, types.ID(jiraURL)),
+	}, nil, nil
+}
+
+func (p *Plugin) submitCreateCloudOAuthInstance(f *flow.Flow, submission map[string]interface{}) (flow.Name, flow.State, map[string]string, error) {
+	jiraURL, _ := submission["url"].(string)
+	if jiraURL == "" {
+		return "", nil, nil, errors.New("no Jira cloud URL in the request")
+	}
+	jiraURL = strings.TrimSpace(jiraURL)
+	if jiraOrgRegexp.MatchString(jiraURL) {
+		jiraURL = fmt.Sprintf("https://%s.atlassian.net", jiraURL)
+	}
+
+	clientID, _ := submission["client_id"].(string)
+	if clientID == "" {
+		return "", nil, nil, errors.New("no Jira OAuth Client ID in the request")
```

Comment:
Check this everywhere
```suggestion
		return "", nil, nil, errors.New("no Jira OAuth Client ID present in the request")
```